src/java/org/apache/cassandra/cql3/restrictions/StatementRestrictions.java

```diff
@@ -266,6 +278,8 @@ else if (relation.isLIKE())
         // there is restrictions not covered by the PK.
         if (!nonPrimaryKeyRestrictions.isEmpty())
         {
+            var columnRestrictions = allColumnRestrictions(clusteringColumnsRestrictions, nonPrimaryKeyRestrictions);
```

Comment:
This seems unused

Reply:
I have refactored this block a bit to handle the validation a bit better. Where possible I have added extra tests as well.